src/Raven.Server/Json/BlittableJsonTextWriterExtensions.cs

```diff
@@ -1642,7 +1638,7 @@ await foreach (var document in documents.WithCancellation(token))
 
         public static void WriteDocument(this AbstractBlittableJsonTextWriter writer, JsonOperationContext context, Document document, bool metadataOnly, Func<LazyStringValue, bool> filterMetadataProperty = null)
         {
-            if (document == null)
+            if (document?.Data == null)
```

Comment:
this looks like a serious change, what is going on here?

Reply:
I need to add the missing included document IDs in the format of: `{"Missing/Included/Id": null, "Included/Doc/Id": {........}}`.
Since this is internal only for sharding, I didn't want to add it to add another property to `QueryResultBase` which is also available on the client.

Reply:
cc @arekpalinski thoughts?

Reply:
Can we have some extra flag in `NonPersistentDocumentFlags` and only then allow `document.Data` to be null?

Reply:
done